static/app/components/organizations/pageFilters/utils.tsx

```diff
@@ -120,10 +124,85 @@ export function isSelectionEqual(selection: PageFilters, other: PageFilters): bo
   return true;
 }
 
+type ProjectId = string | number;
+type EnvironmentId = Environment['id'];
+
+type UpdateData = {
+  project?: ProjectId[] | null;
+  environment?: EnvironmentId[] | null;
+};
+
+/**
+ * Updates the localstorage page filters data
+ *
+ * e.g. if localstorage is empty, user loads issue details for project "foo"
+ * this should not consider "foo" as last used and should not save to local
+ * storage.
+ *
+ * However, if user then changes environment, it should...? Currently it will
+ * save the current project alongside environment to local storage. It's
+ * debatable if this is the desired behavior.
+ *
+ * This will be a no-op if a inaccessible organization slug is passed.
+ */
+export function setPageFiltersStorage(
+  orgSlug: string | null,
+  current: PageFilters,
+  update: UpdateData
+) {
+  const org = orgSlug && OrganizationsStore.get(orgSlug);
+
+  // Do nothing if no org is loaded or user is not an org member. Only
+  // organizations that a user has membership in will be available via the
+  // organizations store
+  if (!org) {
+    return;
+  }
+
+  const {project, environment} = update;
+  const validatedProject = project?.map(Number);
```

Comment:
Do you need to worry about NaN in here?

Reply:
I think we're safe. The types were wrong for some of this before because we don't correctly type our reflux action's, so I think previously it would _always_ be a number (even though there was some logic in there checking if it was a string)

Now I have this matching the usage here https://github.com/getsentry/sentry/blob/69b813d5e63457457cf07a718a0b28ef94748369/static/app/actionCreators/pageFilters.tsx#L28-L29

Just to be safe though, I'll go ahead and filter out nans